File: frontend/passkeys/ios/PasskeysModule.swift
import ExpoModulesCore

public class PasskeysModule: Module {
  let passkeyManager = PasskeyManager()
  
  public func definition() -> ModuleDefinition {
    Name("Passkeys")
    
    AsyncFunction("createPasskeyCredential") { (challengeBase64: String, userIdBase64: String, userName: String, rpId: String, promise: Promise) in
      self.passkeyManager.createPasskeyCredential(challengeBase64: challengeBase64, userIdBase64: userIdBase64, userName: userName, rpId: rpId, promise: promise)
    }
    
    AsyncFunction("signWithPasskeyCredential") { (challengeBase64: String, allowedCredentialsBase64: [String], rpId: String, promise: Promise) in
      self.passkeyManager.signWithPasskeyCredential(challengeBase64: challengeBase64, allowedCredentials: allowedCredentialsBase64, rpId: rpId, promise: promise)
    }
  }
}

File: frontend/passkeys/android/src/main/java/expo/modules/passkeys/PasskeysModule.kt
package expo.modules.passkeys

import android.app.Activity
import android.content.Context
import expo.modules.kotlin.Promise
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PasskeysModule : Module() {
    private val context: Context
        get() = requireNotNull(appContext.reactContext)

    private val currentActivity: Activity?
        get() = appContext.currentActivity

    private val passkeyManager: PasskeyManager by lazy {
        PasskeyManager(context, currentActivity)
    }

    override fun definition() = ModuleDefinition {
        Name("Passkeys")

        AsyncFunction("createPasskeyCredential") { requestJson: String, promise: Promise ->
            passkeyManager.createPasskeyCredential(requestJson, promise)
        }

        AsyncFunction("signWithPasskeyCredential") { requestJson: String, promise: Promise ->
            passkeyManager.signWithPasskeyCredential(requestJson, promise)
        }
    }
}
